flutterproject/fileopener.go

```diff
@@ -0,0 +1,29 @@
+package flutterproject
+
+import (
+	"io"
+	"os"
+)
+
+type FileOpener interface {
+	OpenFile(pth string) (io.Reader, error)
+}
+
+type fileOpener struct {
+}
+
+func NewFileOpener() FileOpener {
+	return fileOpener{}
+}
+
+func (o fileOpener) OpenFile(pth string) (io.Reader, error) {
```

Comment:
This was the previous implementation:

```
	f, err := o.opener.Open(pth)
	if err != nil && !os.IsNotExist(err) {
		return nil, err
	}
	return f, nil
```

The issue with this implementation was that, if there was an `os.IsNotExist` error, the function returned a new `io.Reader` with an underlying nil `os.File` value. If you check if such an interface is nil, it will return false.

Later the code checked if the returned reader is nil and if not it tried to use it, which caused an error. For example here: https://github.com/bitrise-steplib/bitrise-step-flutter-installer/pull/32/files#diff-f85c7f76c4b1019e47803120d423af9a4e1454f949aa0377db0c1d39edca23e0R31-R38